packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,311 @@
+        _contentSensitivitySetting.contentSensitivityBasedOnWidgetCounts!;
+    if (contentSensitivityToRestore != calculatedContentSensitivityLevel) {
+      // Set content sensitivity level as contentSensitivityToRestore. If the call to set content
+      // sensitivity on the platform side fails, then we do not update the current content
+      // sensitivity level.
+      _sensitiveContentService.setContentSensitivity(contentSensitivityToRestore);
+
+      // Update current content sensitivity level.
+      calculatedContentSensitivityLevel = contentSensitivityToRestore;
+    }
+  }
+}
+
+/// Widget to set the [ContentSensitivity] level of content in the widget
+/// tree.
+///
+/// The [sensitivityLevel] of the widget in conjunction with the other
+/// [SensitiveContent] widgets in the tree will determine whether or not the
+/// screen will be obscured during media projection, e.g. screen sharing.
+///
+/// {@macro flutter.services.ContentSensitivity}
+///
+/// Currently, this widget is only supported on Android API 35+. On all lower Android
+/// versions and non-Android platforms, this does nothing; the screen will never be
+/// obscured regardless of the [sensitivityLevel] set. To programmatically check if
+/// a device supports this widget, call [SensitiveContentService.isSupported].
+///
+/// See also:
+///
+///  * [ContentSensitivity], which are the different content sensitivity levels that a
+///    [SensitiveContent] widget can set.
+///
+/// This widget is marked visible for testing because it is not ready for use. It
+/// will be ready for use when it is implemented such that we can guarantee that no
+/// frames will be dropped when content is marked sensitive, ensuring that no sensitive
+/// content will be revealed during media projection.
+// TODO(camsim99): Fix `SensitiveContent` implementation to prevent revealing sensitive
+// content during media projection. Then, export this file to make the widget available
+// for use. See https://github.com/flutter/flutter/issues/160050 and
+// https://github.com/flutter/flutter/issues/164820.
+@visibleForTesting
+class SensitiveContent extends StatefulWidget {
+  /// Creates a [SensitiveContent] widget.
+  const SensitiveContent({super.key, required this.sensitivityLevel, required this.child});
+
+  /// The sensitivity level that the [SensitiveContent] widget should sets for the
+  /// Android native `View` hosting the widget tree.
+  final ContentSensitivity sensitivityLevel;
+
+  /// The child widget of this [SensitiveContent].
+  ///
+  /// If the [sensitivityLevel] is set to [ContentSensitivity.sensitive], then
+  /// the entire screen will be obscured when the screen is projected irrespective
+  /// to the parent/child widgets.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  @override
+  State<SensitiveContent> createState() => _SensitiveContentState();
+}
+
+class _SensitiveContentState extends State<SensitiveContent> {
```

Comment:
Where do we handle the cases for OffStage and Navigator?

Reply:
https://github.com/flutter/flutter/issues/160051 is not addressed in this PR. It will be a follow up.